scripts/core/editor3/helpers/highlights.js

```diff
@@ -856,14 +856,31 @@ function addCommentsForServer(editorState) {
  */
 export function handleBeforeInputHighlights(onChange, chars, editorState) {
     const selection = editorState.getSelection();
-    let {block, newOffset} = getBlockAndOffset(editorState, selection, -1);
 
-    if (block == null) {
-        ({block, newOffset} = getBlockAndOffset(editorState, selection, 1));
+    const {block: blockLeft, newOffset: newOffsetLeft} = getBlockAndOffset(editorState, selection, -1);
+    const {block: blockRight, newOffset: newOffsetRight} = getBlockAndOffset(editorState, selection, 1);
 
-        if (block == null) {
-            return 'not-handled';
-        }
+    const highlightsOnLeftExist = blockLeft != null
+        && blockLeft.getInlineStyleAt(newOffsetLeft).find(styleNameBelongsToHighlight) != null;
+
+    const highlightsOnRightExist = blockRight != null
+        && blockRight.getInlineStyleAt(newOffsetRight).find(styleNameBelongsToHighlight) != null;
+
+    if (highlightsOnLeftExist && highlightsOnRightExist) {
```

Comment:
Is it working well in the case of two different adjacent highlights?

Reply:
It isn't. Thanks for mentioning it, I'll submit a fix for that.